wire-ios-request-strategy/Sources/Request Strategies/Conversation/Actions/UpdateAccessRolesActionHandler.swift

```diff
@@ -51,19 +51,24 @@ final class UpdateAccessRolesActionHandler: ActionHandler<UpdateAccessRolesActio
               }
 
         switch apiVersion {
+
         case .v0:
             return ZMTransportRequest(path: "/conversations/\(conversationID)/access",
                                       method: .put,
                                       payload: payloadAsString as ZMTransportData?,
                                       apiVersion: apiVersion.rawValue)
+
         case .v1, .v2, .v3, .v4, .v5, .v6:
-            guard let domain = conversation.domain.nonEmptyValue ?? BackendInfo.domain else { return nil }
-            return ZMTransportRequest(path: "/conversations/\(domain)/\(conversationID)/access",
-                                      method: .put,
-                                      payload: payloadAsString as ZMTransportData?,
-                                      apiVersion: apiVersion.rawValue)
+            let domain = conversation.domain?.isEmpty == false ? conversation.domain! : BackendInfo.domain
```

Comment:
Could we avoid force unwrapping a` conversation.domain`?

Reply:
I think the best would be to make sure when we set the `domain` to make sure it is not empty and if it is, set it to nil. Then we could skip all the validations on reading, but not sure if this is feasible to refactor now.

Alternatively I think this could be a valid case for an extension `var: resolvedDomain: String`, how is your opinion?

Reply:
My opinion is that we shouldn't use `Optional` if the actual type already has a proper special value. In this case an empty string could be our way to express "no domain". Then these helpers converting empty string to nil wouldn't be required.